# Variables
* Dependent   -   represent with y , aslo Known as 'response variables' or 'target variables'.
* Independent -   represent with x , also known as 'features' or 'regressors'.

# linear equation
* y = mx + c
* m - Slope
* c - intercept

## The r squared value

### Sometimes the points may not be exactly on straight line. There will be deviations from the straight line. This is called error 'E'. The linear regression should consider this error also.


y = ax + b + E

In [2]:
y = [1,2,3,4,5]

In [4]:
# y values are deviated
y1 = [0.8,2.5,3,4.8,4.4]

# Formula
* r squared value = 1 - (sum of E1^2 /sum of E^2)
* E1 = y - y1
* E1^2
* E2 = y - mean
* E2^2
* mean = sum of y values

## Note points
* if r squared value close to 1 then actual value and predicted are very close(High accurancy)
* If it close to 0 Prediction may not be correct.
* If no deviations between y and y1. it means E1 and E1^2 value be 0. This indiactes 100% accurancy.

In [6]:
# import libraries
from sklearn.metrics import r2_score
y = [1,2,3,4,5]
y1 = [0.8,2.5,3,4.8,4.4]
r2_score(y,y1)
print('Coefficient of Determination',r2_score(y,y1))

Coefficient of Determination 0.871


In percentage 0.871 * 100 = 87.1 %
This indicates 87% accurancy level for the model.